packages/textInput/components/TextInputWithButtons.tsx

```diff
@@ -126,18 +146,22 @@ const TextInputWithButtons = (props: TextInputWithButtonsProps) => {
   };
 
   const containerProps: { className?: string } = {};
-  const appearance = getInputAppearance();
-  const dataCy = `textInput textInput.${appearance}`;
+  const calculatedAppearance = getInputAppearance();
+  const dataCy = `textInput textInput.${calculatedAppearance}`;
 
   if (props.className) {
     containerProps.className = props.className;
   }
   return (
     <div {...containerProps} data-cy={dataCy}>
       {renderLabel({
-        appearance,
-        hidden: !props.showInputLabel,
-        id: getId(props),
+        appearance: calculatedAppearance,
+        hidden: !showInputLabel,
+        id: getId({
+          type,
+          appearance,
+          showInputLabel
+        }),
```

Comment:
This function is looking for a prop of `id` so I think what we need to pass in `props` here instead.

```suggestion
        id: getId(props),
```

Reply:
Good call, thanks for the guidance! 🙌